miden-lib/src/transaction/procedures.rs

```diff
@@ -0,0 +1,223 @@
+use alloc::vec::Vec;
+
+use miden_objects::{Digest, Felt, Hasher};
+
+use super::TransactionKernel;
+
+impl TransactionKernel {
+    /// Hashes of all dynamically executed kernel procedures.
+    pub const PROCEDURES: [Digest; 28] = [
+        // get_account_id
+        Digest::new([
+            Felt::new(8040261465733444704),
+            Felt::new(11111141085375373880),
+            Felt::new(7423929485586361344),
+            Felt::new(4119214601469502087),
+        ]),
```

Comment:
I wonder how difficult would it be to write something like a `digest!` macro which would basically do this. And then, we'd be able to do something like:
```rust
digest!(0x123455...)
```

Reply:
I was thinking of something like this too, but the problem is that `PROCEDURES` array is a constant array, so I can't use non-constant methods during its creation, only `Digest::new`. Theoretically I can try to move the body of the `hex_to_bytes` crypto procedure to the macro, but I think that it is an overkill. 
But I think that we can write a macro which would do the same but will take just four elements — that will make the future updates easier.